### PR TITLE
Add delegating handler exposure to ClientFactory

### PR DESCRIPTION
**Pull request by jianghaolu**
Need to add tests

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@jianghaolu__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

**Comment by jianghaolu** [comment]
@amarzavery

**Comment by jianghaolu** [comment]
@azuresdkci add to whitelist

File: src/Authentication/Common.Authentication/Factories/ClientFactory.cs [hunk 1]
```diff
@@ -91,6 +94,10 @@ public ClientFactory()
 
             SubscriptionCloudCredentials creds = AzureSession.AuthenticationFactory.GetSubscriptionCloudCredentials(context, endpoint);
             TClient client = CreateCustomClient<TClient>(creds, context.Environment.GetEndpointAsUri(endpoint));
+            foreach(DelegatingHandler handler in handlers.Values.ToArray())
+            {
+                client = client.WithHandler(handler);
```
Comment (ogail) [inline src/Authentication/Common.Authentication/Factories/ClientFactory.cs]: Feel free to use `client.AddHandlerToPipeline(handler)`

File: src/Authentication/Common.Authentication/Factories/ClientFactory.cs [hunk 2]
```diff
@@ -236,6 +243,19 @@ public void RemoveAction(Type actionType)
             }
         }
 
+        public void AddHandler(DelegatingHandler handler)
+        {
+            handlers[handler.GetType()] = handler;
```
Comment (ogail) [inline src/Authentication/Common.Authentication/Factories/ClientFactory.cs]: add if statement that handler is not null
```csharp
if (handler != null)
{
    handlers[handler.GetType()] = handler;
}
```

**Pull request merged**